### PR TITLE
add latest docu of blockly features of M2

### PR DESCRIPTION
**Pull request by stefan-hoehn**
This adds missing documentation for the following new blockly features

- https://github.com/openhab/openhab-webui/pull/2175 Add parse number block
- https://github.com/openhab/openhab-webui/pull/2174 persistence previous state value as number
- https://github.com/openhab/openhab-webui/pull/2411 HTTP request block
- https://github.com/openhab/openhab-webui/pull/2465 Add undefined block 
- https://github.com/openhab/openhab-webui/pull/2562 New Thing blocks
- https://github.com/openhab/openhab-webui/pull/2564 breaking change persistence 
- https://github.com/openhab/openhab-webui/pull/2575 add timer context 

and many more small corrections

### CODE REVIEW

File: configuration/blockly/rules-blockly-items-things.md [hunk 1]
```diff
@@ -303,6 +318,35 @@ Function: Gets a **Thing Status** for use in other Thing related functions
...
+- status info (status plus further info like "light not reachable"): String
+- location: String
+- thing type UID (unique ID of the Thing type): String
+- bridge UID (unique ID of the Bridge): String
+
+### Get Things
+
+Function: Gets all things and returns a list of [Thing-Objects](#get-thing).
+
+![blockly-getItemAttributes](../images/blockly/blockly-get-things.png)
```
Comment (florian-h05) [inline configuration/blockly/rules-blockly-items-things.md:343]:
```suggestion
![blockly-getThings](../images/blockly/blockly-get-things.png)
```

File: configuration/blockly/rules-blockly-items-things.md [hunk 2]
```diff
@@ -303,6 +318,35 @@ Function: Gets a **Thing Status** for use in other Thing related functions
   - REMOVING
   - REMOVED
 
+### Get particular attributes of a Thing
+
+![blockly-getItemAttributes](../images/blockly/blockly-getThingAttributes.png)
```
Comment (florian-h05) [inline configuration/blockly/rules-blockly-items-things.md:323]:
```suggestion
![blockly-getThingAttributes](../images/blockly/blockly-getThingAttributes.png)
```

File: configuration/blockly/rules-blockly-items-things.md [hunk 3]
```diff
@@ -303,6 +318,35 @@ Function: Gets a **Thing Status** for use in other Thing related functions
...
+### Get Things
+
+Function: Gets all things and returns a list of [Thing-Objects](#get-thing).
+
+![blockly-getItemAttributes](../images/blockly/blockly-get-things.png)
+
+The following loop iterates over all things and logs out the status.
+Note that it uses a [typed variable](./index.html#using-variables) for that.
+
+![blockly-getItemAttributes](../images/blockly/blockly-things-loop.png)
```
Comment (florian-h05) [inline configuration/blockly/rules-blockly-items-things.md:348]:
```suggestion
![blockly-thingsLoop](../images/blockly/blockly-things-loop.png)
```

File: configuration/blockly/rules-blockly-persistence.md [hunk 4]
```diff
@@ -36,6 +36,18 @@ More about that topic can be viewed at ![youtube](../images/blockly/youtube-logo
...
+Due to adding future values in openHAB the amount of attributes has been vastly increased and the names had to be renamed to become more specific in terms of historic and future states.
+
+**Important:** **Due to a breaking change of the internal methods in openHAB 4.2, Blockly rules that use persistence methods need to be migrated once.**
+This does not happen automatically but needs to be done opening the blockly rule once and re-save it.
+Blockly then automatically rewrites the rule to be compatible.
+
+Values:
+
+The following values are available as _historic_ and _future_ representations.
+
```
Comment (florian-h05) [inline configuration/blockly/rules-blockly-persistence.md:50]:
```suggestion

- persisted state: gets the persisted state at a certain point in time
```

File: configuration/blockly/rules-blockly-standard-ext.md [hunk 5]
```diff
@@ -239,7 +243,7 @@ Example:
 ## Loops
 
 Even though there a no specialized openHAB blocks provided, loops are used rather often.
-Therefore there is a good introduction to loops available which can be viewed at ![youtube](../images/blockly/youtube-logo-small.png) [Loops in Blockly](https://youtu.be/EdllUlJ7p6k?t=1947)
+Therefore, there is a good introduction to loops available which can be viewed at ![youtube](../images/blockly/youtube-logo-small.png) [Loops in Blockly](https://youtu.be/EdllUlJ7p6k?t=1947)
```
Comment (florian-h05) [inline configuration/blockly/rules-blockly-standard-ext.md:246]:
```suggestion
Therefore, there is a good introduction to loops available which can be viewed at ![youtube](../images/blockly/youtube-logo-small.png) [Loops in Blockly](https://youtu.be/EdllUlJ7p6k?t=1947).
```

File: package-lock.json
Comment (florian-h05) [inline package-lock.json:1]: Can you please revert the package-lock changes? This is unrelated to this PR.

File: tutorials/getting_started/rules_scenes.md [hunk 6]
```diff
@@ -101,17 +101,31 @@ When you now click on that label card, the scene rule will be triggered and all
 #### Rule
 
 If we want to use triggers that are known from other rule types we can write such a rule, define that trigger and let that rule call the scene rule.
-For the purpose of the tutorial we use a very simple [Blockly Rule]({{base}}/tutorial/rules-blockly.html).
 See the details there on how to create a trigger (e.g. a cron trigger that will call the blockly rule at 22:00 in the evening) to execute that rule.
+The section below also provides examples for JavaScript and Python.
```
Comment (florian-h05) [inline tutorials/getting_started/rules_scenes.md:105]:
```suggestion
The section below provides examples for Blockly and various rule/script languages.
```

File: tutorials/getting_started/rules_scenes.md [hunk 7]
```diff
@@ -101,17 +101,31 @@ When you now click on that label card, the scene rule will be triggered and all
...
 
 ```javascript
  rules.runRule('scene_office_dimmed_light', {});
 ```
+
+:::
+
+::: tab JRuby
+
+rules[“id”] gives you the rule object for the given id, which supports the [trigger method](https://openhab.github.io/openhab-jruby/main/OpenHAB/Core/Rules/Rule.html#trigger-instance_method) (aliased as run).
```
Comment (florian-h05) [inline tutorials/getting_started/rules_scenes.md:127]:
```suggestion
`rules[“"d"]` gives you the rule object for the given id, which supports the [trigger method](https://openhab.github.io/openhab-jruby/main/OpenHAB/Core/Rules/Rule.html#trigger-instance_method) (aliased as run).
```

File: tutorials/getting_started/rules_scenes.md [hunk 8]
```diff
@@ -101,17 +101,31 @@ When you now click on that label card, the scene rule will be triggered and all
...
+
+> rules["scene_id"].run
```
Comment (florian-h05) [inline tutorials/getting_started/rules_scenes.md:129]: Please use code fences and set `ruby` or `rb` as language.

File: configuration/blockly/rules-blockly-timers-and-delays.md [hunk 9]
```diff
@@ -37,6 +37,34 @@ and cancels the same timer in Rule 2
 
 ![timer-shared-cancel](../images/blockly/blockly-timer-shared-cancel.png)
 
+## Timer context
+
+Timers provide a statement block where the blocks reside that are run when the timer triggers.
+The context allows data to be provided which is later used when the blocks are triggered by the timer (note that in many cases you can just leave the context empty or provide an [undefined block](rules-blockly-standard-ext.md#logic))
```
Comment (florian-h05) [inline configuration/blockly/rules-blockly-timers-and-delays.md:43]:
```suggestion
The context allows data to be provided which is later used when the blocks are triggered by the timer (note that in many cases you can just leave the context empty or provide an [undefined block](rules-blockly-standard-ext.md#logic)).
```

File: configuration/blockly/rules-blockly-timers-and-delays.md [hunk 10]
```diff
@@ -37,6 +37,34 @@ and cancels the same timer in Rule 2
...
+
+To provide a context put any block to the context, which can be a block of any type like String, Number, a variable or even a Dictionary which could hold several values via a key/value mapping.
```
Comment (florian-h05) [inline configuration/blockly/rules-blockly-timers-and-delays.md:45]:
```suggestion
To provide a context put any block to the context input, which can be a block of any type like String, Number, a variable or even a Dictionary which could hold several values via a key/value mapping.
```

File: configuration/blockly/rules-blockly-timers-and-delays.md [hunk 11]
```diff
@@ -37,6 +37,34 @@ and cancels the same timer in Rule 2
...
+
+The context can be used within the timer statement block by utilizing the "timer context" block:
+
+![blockly-timer-context.png](../images/blockly/blockly-timer-context.png)
+
+Here are some examples how this feature can be used
```
Comment (florian-h05) [inline configuration/blockly/rules-blockly-timers-and-delays.md:51]:
```suggestion
Here are some examples how this feature can be used:
```

File: configuration/blockly/rules-blockly-timers-and-delays.md [hunk 12]
```diff
@@ -37,6 +37,34 @@ and cancels the same timer in Rule 2
...
+
+![timer-context-examples](../images/blockly/blockly-timer-context-examples.png)
+
+The loop is in particular interesting to look at:
+
+- The loop counts from 1 to 10 (actually to 9)
```
Comment (florian-h05) [inline configuration/blockly/rules-blockly-timers-and-delays.md:57]:
```suggestion
- The loop counts from 1 to 10 (actually to 9).
```

**Review by florian-h05 — COMMENTED** [review]
In general LGTM, please have a look at my comments.

Please do not make the suggested changes locally and then push a commit, just go to the code tab, add the suggestions "to batch" and then "commit and sign-off batch" (green button). This automatically resolves the suggestions and eases review for us both :+1:

File: tutorials/getting_started/rules_scenes.md (hunk 7 above)
Comment (stefan-hoehn) [reply tutorials/getting_started/rules_scenes.md:127]: I think this isn't intended: `rules[“"d"]`

Comment (florian-h05) [reply tutorials/getting_started/rules_scenes.md:127]: Better now?

File: tutorials/getting_started/rules_scenes.md [hunk 13]
```diff
@@ -101,17 +101,33 @@ When you now click on that label card, the scene rule will be triggered and all
...
+
+:::
+
+::: tab JRuby
+
+rules[“id”] gives you the rule object for the given id, which supports the [trigger method](https://openhab.github.io/openhab-jruby/main/OpenHAB/Core/Rules/Rule.html#trigger-instance_method) (aliased as run).
+
+```ruby
+  rules["scene_id"].run
+``
```
Comment (florian-h05) [inline tutorials/getting_started/rules_scenes.md:131]: There should be three of these signs.

**Review by florian-h05 — COMMENTED** [review]
One last comment:

**Review by florian-h05 — APPROVED** [review]
LGTM, thanks!
